plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java

```diff
@@ -283,7 +291,7 @@ public void validate() {
       return;
     }
 
-    projectInput.setEnabled(true);
+    projectInput.setEnabled(canEnableChildren);
```

Comment:
Could be done instead by expanding 234 to:
```
if (selectedCredential == null || !canEnableChildren) {
```

Reply:
That would make enablement correct at least, but we still want to validate all the global preference values to show proper validation errors even if the panel is locked from editing (disabled).